Tools/cases_generator/analyzer.py

```diff
@@ -0,0 +1,458 @@
+from dataclasses import dataclass
+import lexer
+import parser
+from typing import Optional
+
+
+@dataclass
```

Comment:
I would be tempted to do this, for readability. I agree with Guido's opinion that it's hard to read when instances of this class are constructed using positional arguments

```suggestion
@dataclass(kw_only=True)
```